diff -urN udev-145.orig/configure.ac udev-145/configure.ac
--- udev-145.orig/configure.ac	2009-07-08 06:50:09.000000000 +0930
+++ udev-145/configure.ac	2010-12-12 14:14:16.942358001 +1030
@@ -62,6 +62,10 @@
 	AC_CHECK_LIB([acl], [acl_init], [:], AC_MSG_ERROR([libacl not found]))
 	AC_CHECK_HEADER([acl/libacl.h], [:], AC_MSG_ERROR([libacl header not found]))
 
+	PKG_CHECK_MODULES(LIBUDEV, libudev >= 145)
+	AC_SUBST(LIBUDEV_CFLAGS)
+	AC_SUBST(LIBUDEV_LIBS)
+
 	PKG_CHECK_MODULES(LIBUSB, libusb >= 0.1.12)
 	AC_SUBST(LIBUSB_CFLAGS)
 	AC_SUBST(LIBUSB_LIBS)
diff -urN udev-145.orig/extras/gudev/Makefile.am udev-145/extras/gudev/Makefile.am
--- udev-145.orig/extras/gudev/Makefile.am	2009-07-04 11:42:06.000000000 +0930
+++ udev-145/extras/gudev/Makefile.am	2010-12-12 14:13:41.181358001 +1030
@@ -54,9 +54,9 @@
 	gudevprivate.h \
 	$(BUILT_SOURCES)
 
-libgudev_1_0_la_CFLAGS = $(GLIB_CFLAGS)
+libgudev_1_0_la_CFLAGS = $(LIBUDEV_CFLAGS) $(GLIB_CFLAGS)
 
-libgudev_1_0_la_LIBADD = $(top_builddir)/libudev/libudev.la $(GLIB_LIBS)
+libgudev_1_0_la_LIBADD = $(LIBUDEV_LIBS) $(GLIB_LIBS)
 
 LT_CURRENT=0
 LT_REVISION=1
diff -urN udev-145.orig/extras/Makefile.am udev-145/extras/Makefile.am
--- udev-145.orig/extras/Makefile.am	2009-06-17 19:42:08.000000000 +0930
+++ udev-145/extras/Makefile.am	2010-12-12 14:13:38.295358004 +1030
@@ -1,25 +1,4 @@
 include $(top_srcdir)/Makefile.am.inc
 
-SUBDIRS = \
-	ata_id \
-	cdrom_id \
-	edd_id \
-	path_id \
-	firmware \
-	collect \
-	floppy \
-	fstab_import \
-	rule_generator \
-	scsi_id \
-	usb_id \
-	v4l_id
+SUBDIRS = gudev
 
-if ENABLE_EXTRAS
-SUBDIRS += \
-	udev-acl \
-	usb-db \
-	hid2hci \
-	keymap \
-	modem-modeswitch \
-	gudev
-endif
diff -urN udev-145.orig/Makefile.am udev-145/Makefile.am
--- udev-145.orig/Makefile.am	2009-06-20 02:54:10.000000000 +0930
+++ udev-145/Makefile.am	2010-12-12 14:13:47.563358008 +1030
@@ -1,15 +1,6 @@
 include $(top_srcdir)/Makefile.am.inc
-
-SUBDIRS = \
-	docs \
-	libudev \
-	udev \
-	rules \
-	extras
-
-udevconfdir = $(sysconfdir)/udev
-dist_udevconf_DATA = \
-	udev.conf
+ 
+SUBDIRS = extras
 
 EXTRA_DIST = \
 	autogen.sh
