--- ibus-1.4.99.20120712/conf/dconf/config.c	2012-06-01 11:54:27.000000000 -0400
+++ ibus-1.4.99.20120712.dconf/conf/dconf/config.c	2012-07-17 20:17:19.243579503 -0400
@@ -159,12 +159,14 @@
 _watch_func (DConfClient         *client,
              const gchar         *gpath,
              const gchar * const *items,
-             gint                 n_items,
              const gchar         *tag,
              IBusConfigDConf     *config)
 {
     gchar **gkeys = NULL;
     gint i;
+    gint n_items;
+
+    n_items = g_strv_length ((gchar **)items);
 
     g_return_if_fail (gpath != NULL);
     g_return_if_fail (n_items >= 0);
@@ -231,16 +233,13 @@
 ibus_config_dconf_init (IBusConfigDConf *config)
 {
     GVariant *variant;
-    GError *error;
 
-    config->client = dconf_client_new ("ibus",
-                                       (DConfWatchFunc)_watch_func,
-                                       config,
-                                       NULL);
-
-    error = NULL;
-    if (!dconf_client_watch (config->client, DCONF_PREFIX"/", NULL, &error))
-        g_warning ("Can not watch dconf path %s", DCONF_PREFIX"/");
+    config->client = dconf_client_new ();
+
+    g_signal_connect (config->client, "changed",
+                      G_CALLBACK (_watch_func), config);
+
+    dconf_client_watch_fast (config->client, DCONF_PREFIX"/");
 
     config->preserve_name_prefixes = NULL;
     variant = dconf_client_read (config->client,
@@ -265,9 +264,7 @@
 ibus_config_dconf_destroy (IBusConfigDConf *config)
 {
     if (config->client) {
-        GError *error = NULL;
-        if (!dconf_client_unwatch (config->client, DCONF_PREFIX"/", NULL, &error))
-            g_warning ("Can not unwatch dconf path %s", DCONF_PREFIX"/");
+        dconf_client_unwatch_fast (config->client, DCONF_PREFIX"/");
 
         g_object_unref (config->client);
         config->client = NULL;
@@ -307,12 +304,7 @@
         g_free (gname);
     }
 
-    retval = dconf_client_write (client,
-                                 gkey,
-                                 value,
-                                 NULL,   /* tag */
-                                 NULL,   /* cancellable */
-                                 error);
+    retval = dconf_client_write_fast (client, gkey, value, error);
     g_free (gkey);
 
     return retval;
--- ibus-1.4.99.20120712/conf/dconf/config.h	2012-06-01 11:54:27.000000000 -0400
+++ ibus-1.4.99.20120712.dconf/conf/dconf/config.h	2012-07-17 20:15:20.182583759 -0400
@@ -23,7 +23,8 @@
 #define __CONFIG_DCONF_H__
 
 #include <ibus.h>
-#include <dconf/dconf.h>
+#include <client/dconf-client.h>
+#include <common/dconf-paths.h>
 
 #define IBUS_TYPE_CONFIG_DCONF            \
     (ibus_config_dconf_get_type ())
