diff -ur libproxy-0.4.3.orig/bindings/CMakeLists.txt libproxy-0.4.3/bindings/CMakeLists.txt
--- libproxy-0.4.3.orig/bindings/CMakeLists.txt	2010-06-08 01:27:23.000000000 +0930
+++ libproxy-0.4.3/bindings/CMakeLists.txt	2010-06-10 22:27:04.147396632 +0930
@@ -5,10 +5,10 @@
   if(PYTHONINTERP_FOUND)
     execute_process(COMMAND
 		    ${PYTHON_EXECUTABLE}
-		    -c "import distutils.sysconfig ; print distutils.sysconfig.get_python_lib(plat_specific=0)"
-		    OUTPUT_VARIABLE pysitepkgdir
+		    -c "import distutils.sysconfig ; print distutils.sysconfig.get_python_version()"
+		    OUTPUT_VARIABLE pyversion
 		    OUTPUT_STRIP_TRAILING_WHITESPACE)
-    install(FILES libproxy.py DESTINATION ${pysitepkgdir})
+    install(FILES libproxy.py DESTINATION ${libdir}/python${pyversion}/site-packages)
   endif()
 endif()
 
diff -ur libproxy-0.4.3.orig/bindings/perl/CMakeLists.txt libproxy-0.4.3/bindings/perl/CMakeLists.txt
--- libproxy-0.4.3.orig/bindings/perl/CMakeLists.txt	2010-06-08 01:27:23.000000000 +0930
+++ libproxy-0.4.3/bindings/perl/CMakeLists.txt	2010-06-10 22:27:04.147396632 +0930
@@ -11,8 +11,9 @@
   else()
     # PERL_SITEARCH is actually not defined in latest FindPerlLibs.cmake macros, thus
     # we have to find the correct path on our own.
-    EXECUTE_PROCESS(COMMAND ${PERL_EXECUTABLE} -MConfig -e "print \$Config{sitearch}"
-      OUTPUT_VARIABLE PX_PERL_ARCH)
+    EXECUTE_PROCESS(COMMAND ${PERL_EXECUTABLE} -MConfig -e "print \$Config{PERL_REVISION}"
+      OUTPUT_VARIABLE PX_PERL_REVISION)
+    set (PX_PERL_ARCH ${libdir}/perl${PX_PERL_REVISION})
     set (PX_PERL_LIB ${PERL_SITELIB})
   endif()
 
