--- libtool-1.5/libtool.m4.x86_64	2003-04-14 17:58:23.000000000 -0400
+++ libtool-1.5/libtool.m4	2003-04-24 16:38:34.000000000 -0400
@@ -2167,7 +2167,7 @@
 # This must be Linux ELF.
 linux*)
   case $host_cpu in
-  alpha*|hppa*|i*86|ia64*|m68*|mips*|powerpc*|sparc*|s390*|sh*)
+  alpha*|hppa*|i*86|ia64*|m68*|mips*|powerpc*|sparc*|s390*|sh*|x86_64*)
     lt_cv_deplibs_check_method=pass_all ;;
   *)
     # glibc up to 2.1.1 does not perform some relocations on ARM
