https://bugs.launchpad.net/mir/+bug/1323317

diff -u -r mir-0.1.9-orig/src/server/report/CMakeLists.txt mir-0.1.9/src/server/report/CMakeLists.txt
--- mir-0.1.9-orig/src/server/report/CMakeLists.txt	2014-05-26 15:56:47.624556441 +0200
+++ mir-0.1.9/src/server/report/CMakeLists.txt	2014-05-26 15:56:57.128675452 +0200
@@ -7,4 +7,4 @@
     default_server_configuration.cpp
     )
 
-target_link_libraries(mirreport mirserverlttng mirlogging mirnullreport)
+target_link_libraries(mirreport mirlogging mirnullreport)
