diff -urN nss-3.12.6.orig/autogen.sh nss-3.12.6/autogen.sh
--- nss-3.12.6.orig/autogen.sh	1970-01-01 09:30:00.000000000 +0930
+++ nss-3.12.6/autogen.sh	2010-11-23 21:54:08.627387993 +1030
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+test -n "$srcdir" || srcdir=`dirname "$0"`
+test -n "$srcdir" || srcdir=.
+
+olddir=`pwd`
+cd "$srcdir"
+
+autoreconf --force --install || exit $?
+
+cd "$olddir"
+test -n "$NOCONFIGURE" || "$srcdir/configure" "$@"
+
diff -urN nss-3.12.6.orig/configure.ac nss-3.12.6/configure.ac
--- nss-3.12.6.orig/configure.ac	1970-01-01 09:30:00.000000000 +0930
+++ nss-3.12.6/configure.ac	2010-11-23 21:53:58.668388009 +1030
@@ -0,0 +1,24 @@
+AC_PREREQ(2.54)
+AC_INIT([nss], [3.12])
+
+AC_CONFIG_SRCDIR([mozilla/security/nss/Makefile])
+
+NSPR_PREFIX=$(pkg-config nspr --variable=prefix)
+AC_SUBST(NSPR_PREFIX)
+NSPR_LIBDIR=$(pkg-config nspr --variable=libdir)
+AC_SUBST(NSPR_LIBDIR)
+
+AC_CHECK_SIZEOF(long)
+if test x$ac_cv_sizeof_long = x8; then
+  USE_64=1
+else
+  USE_64=0
+fi
+AC_SUBST(USE_64)
+
+AC_CONFIG_FILES([
+Makefile
+nss.pc
+])
+
+AC_OUTPUT
diff -urN nss-3.12.6.orig/Makefile.in nss-3.12.6/Makefile.in
--- nss-3.12.6.orig/Makefile.in	1970-01-01 09:30:00.000000000 +0930
+++ nss-3.12.6/Makefile.in	2010-11-23 21:53:58.668388009 +1030
@@ -0,0 +1,36 @@
+prefix=@prefix@
+nspr_prefix=@NSPR_PREFIX@
+nspr_libdir=@NSPR_LIBDIR@
+libdir=@libdir@
+
+ARGS = \
+	MOZILLA_CLIENT=1 \
+	NSPR_INCLUDE_DIR=$(nspr_prefix)/include/nspr \
+	NSPR_LIB_DIR=$(nspr_libdir) \
+	SOURCE_MD_DIR=$(prefix) \
+	DIST=$(prefix) \
+	BUILD_OPT=1 \
+	NS_USE_GCC=1 \
+	NSS_USE_SYSTEM_SQLITE=1 \
+	NSDISTMODE=copy \
+	USE_64=@USE_64@ \
+	NSS_ENABLE_ECC=1
+
+all:
+	$(MAKE) -C mozilla/security/nss \
+		build_coreconf \
+		build_dbm \
+		all \
+		$(ARGS)
+
+clean:
+	$(MAKE) -C mozilla/security/nss clean
+
+install:
+	$(MAKE) -C mozilla/security/nss install $(ARGS)
+	-mkdir $(prefix)/include/nss
+	install -m 644 -t $(prefix)/include/nss mozilla/dist/public/nss/*
+	-rm $(prefix)/lib/libssl.a
+	-mkdir -p $(libdir)/pkgconfig/
+	cp nss.pc $(libdir)/pkgconfig/nss.pc
+
diff -urN nss-3.12.6.orig/nss.pc.in nss-3.12.6/nss.pc.in
--- nss-3.12.6.orig/nss.pc.in	1970-01-01 09:30:00.000000000 +0930
+++ nss-3.12.6/nss.pc.in	2010-11-23 21:53:58.668388009 +1030
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include/nss
+
+Name: NSS
+Description: Mozilla Network Security Services
+Version: 3.12
+Requires: nspr
+Libs: -L${libdir} -lnss3 -lnssutil3 -lsmime3 -lssl3
+Cflags: -I${includedir}
