diff -ur samba-4.0.0alpha11.orig/lib/talloc/Makefile.in samba-4.0.0alpha11/lib/talloc/Makefile.in
--- samba-4.0.0alpha11.orig/lib/talloc/Makefile.in	2010-01-11 06:35:28.000000000 +0100
+++ samba-4.0.0alpha11/lib/talloc/Makefile.in	2010-10-12 19:51:19.548136159 +0200
@@ -56,7 +56,7 @@
 
 distclean:: clean
 	rm -f Makefile
-	rm -f config.log config.status config.h config.cache
+	rm -f config.log config.status config.h config.cache talloc.pc
 
 realdistclean:: distclean
 	rm -f configure config.h.in
