diff -ur samba-4.0.0alpha11.orig/lib/tdb/Makefile.in samba-4.0.0alpha11/lib/tdb/Makefile.in
--- samba-4.0.0alpha11.orig/lib/tdb/Makefile.in	2010-01-11 06:35:28.000000000 +0100
+++ samba-4.0.0alpha11/lib/tdb/Makefile.in	2010-10-12 19:51:07.045949157 +0200
@@ -72,7 +72,7 @@
 	rm -fr abi
 
 distclean:: clean
-	rm -f config.log config.status include/config.h config.cache
+	rm -f config.log config.status include/config.h config.cache tdb.pc
 	rm -f Makefile
 
 realdistclean:: distclean
diff -ur samba-4.0.0alpha11.orig/lib/tdb/tdb.mk samba-4.0.0alpha11/lib/tdb/tdb.mk
--- samba-4.0.0alpha11.orig/lib/tdb/tdb.mk	2010-01-11 06:35:28.000000000 +0100
+++ samba-4.0.0alpha11/lib/tdb/tdb.mk	2010-10-12 19:51:09.625987747 +0200
@@ -38,7 +38,7 @@
 clean:: 
 	rm -f test.db test.tdb torture.tdb test.gdbm
 	rm -f $(TDB_SONAME) $(TDB_SOLIB) $(TDB_STLIB) libtdb.$(SHLIBEXT)
-	rm -f $(ALL_PROGS) tdb.pc
+	rm -f $(ALL_PROGS)
 	rm -f tdb.exports.sort tdb.exports.check tdb.exports.check.sort
 	rm -f tdb.signatures.sort tdb.signatures.check tdb.signatures.check.sort
 
