diff -ur samba-4.0.0alpha11.orig/lib/tevent/Makefile.in samba-4.0.0alpha11/lib/tevent/Makefile.in
--- samba-4.0.0alpha11.orig/lib/tevent/Makefile.in	2010-01-11 06:35:28.000000000 +0100
+++ samba-4.0.0alpha11/lib/tevent/Makefile.in	2010-10-12 19:51:27.661257521 +0200
@@ -72,7 +72,7 @@
 	rm -fr abi
 
 distclean:: clean
-	rm -f config.log config.status config.h config.cache
+	rm -f config.log config.status config.h config.cache tevent.pc
 	rm -f Makefile
 
 realdistclean:: distclean
diff -ur samba-4.0.0alpha11.orig/lib/tevent/tevent.mk samba-4.0.0alpha11/lib/tevent/tevent.mk
--- samba-4.0.0alpha11.orig/lib/tevent/tevent.mk	2010-01-11 06:35:28.000000000 +0100
+++ samba-4.0.0alpha11/lib/tevent/tevent.mk	2010-10-12 19:51:30.672302559 +0200
@@ -41,6 +41,5 @@
 
 clean::
 	rm -f $(TEVENT_SOBASE) $(TEVENT_SONAME) $(TEVENT_SOLIB) $(TEVENT_STLIB)
-	rm -f tevent.pc
 	rm -f tevent.exports.sort tevent.exports.check tevent.exports.check.sort
 	rm -f tevent.signatures.sort tevent.signatures.check tevent.signatures.check.sort
