Fix WAF to work as an autotools module type.

--- waf-1.4.3/wscript	2008-05-20 15:32:51 +0000
+++ waf-1.4.3/wscript	2008-05-20 15:33:28 +0000
@@ -314,9 +314,6 @@
 			sys.exit(0)
 
 	if Params.g_commands['install']:
-		val = Params.g_options.yes or (not sys.stdin.isatty() or raw_input("Installing Waf is discouraged. Proceed? [y/n]"))
-		if val != True and val != "y": sys.exit(1)
-
 		compute_revision()
 
 		create_waf()
--- waf-1.4.3/utils/autogen.sh	2008-06-23 01:52:48 +0200
+++ waf-1.4.3/utils/autogen.sh	2008-08-21 13:52:04 +0200
@@ -47,3 +47,5 @@
 
 EOF
 
+./configure --prefix=$prefix
+./waf configure $prefix
