Fix stupid prompt for WAF installation.

--- waf-1.4.2/wscript	2008-05-20 15:32:51 +0000
+++ waf-1.4.2/wscript	2008-05-20 15:33:28 +0000
@@ -314,9 +314,6 @@
 			sys.exit(0)
 
 	if Params.g_commands['install']:
-		val = Params.g_options.yes or (not sys.stdin.isatty() or raw_input("Installing Waf is discouraged. Proceed? [y/n]"))
-		if val != True and val != "y": sys.exit(1)
-
 		compute_revision()
 
 		create_waf()
