Index: /trunk/Source/WebCore/dom/Document.idl
===================================================================
--- /trunk/Source/WebCore/dom/Document.idl	(revision 125182)
+++ /trunk/Source/WebCore/dom/Document.idl	(revision 125183)
@@ -76,5 +76,5 @@
             raises (DOMException);
 
-#if defined(LANGUAGE_OBJECTIVE_C) && LANGUAGE_OBJECTIVE_C
+#if defined(LANGUAGE_OBJECTIVE_C) && LANGUAGE_OBJECTIVE_C || defined(LANGUAGE_GOBJECT) && LANGUAGE_GOBJECT
         // document.documentURI was writable in DOM3 Core, but is read-only in DOM4
         // (see http://www.w3.org/TR/2011/WD-dom-20110915/#document). We need to keep
Index: /trunk/Source/WebCore/ChangeLog
===================================================================
--- /trunk/Source/WebCore/ChangeLog	(revision 125182)
+++ /trunk/Source/WebCore/ChangeLog	(revision 125183)
@@ -1,2 +1,15 @@
+2012-08-09  Xan Lopez  <xlopez@igalia.com>
+
+        1.9.6 drops symbols, breaking compatibility
+        https://bugs.webkit.org/show_bug.cgi?id=93477
+
+        Reviewed by Carlos Garcia Campos.
+
+        Expose documentURI as writable again for the GObject DOM
+        bindings. DOM4 makes it read-only, but we need to revert to the
+        old state for compatibility reasons, just like Objective C.
+
+        * dom/Document.idl: ditto.
+
 2012-08-09  Mario Sanchez Prada  <msanchez@igalia.com>
 
