Index: /trunk/WebCore/plugins/gtk/gtk2xtbin.c
===================================================================
--- /trunk/WebCore/plugins/gtk/gtk2xtbin.c	(revision 70677)
+++ /trunk/WebCore/plugins/gtk/gtk2xtbin.c	(revision 73603)
@@ -275,18 +275,23 @@
   GtkAllocation widget_allocation;
 #endif
+
+#ifdef DEBUG_XTBIN
+  printf("gtk_xtbin_realize()\n");
+#endif
+
+  g_return_if_fail (GTK_IS_XTBIN (widget));
+
+  xtbin = GTK_XTBIN (widget);
+
+  /* caculate the allocation before realize */
+#if GTK_CHECK_VERSION(2, 24, 0)
+  allocation.width = gdk_window_get_width(xtbin->parent_window);
+  allocation.height = gdk_window_get_height(xtbin->parent_window);
+#else
   gint  x, y, w, h, d; /* geometry of window */
-
-#ifdef DEBUG_XTBIN
-  printf("gtk_xtbin_realize()\n");
-#endif
-
-  g_return_if_fail (GTK_IS_XTBIN (widget));
-
-  xtbin = GTK_XTBIN (widget);
-
-  /* caculate the allocation before realize */
   gdk_window_get_geometry(xtbin->parent_window, &x, &y, &w, &h, &d);
   allocation.width = w;
   allocation.height = h;
+#endif
   gtk_widget_size_allocate (widget, &allocation);
 
Index: /trunk/WebCore/ChangeLog
===================================================================
--- /trunk/WebCore/ChangeLog	(revision 73602)
+++ /trunk/WebCore/ChangeLog	(revision 73603)
@@ -1,2 +1,10 @@
+2010-12-09  Gustavo Noronha Silva  <gustavo.noronha@collabora.co.uk>
+
+        Build fix for GTK+3. Use functions added for GTK+ 2.24 to get the
+        GDK window size.
+
+        * plugins/gtk/gtk2xtbin.c:
+        (gtk_xtbin_realize):
+
 2010-12-09  Siddharth Mathur  <siddharth.mathur@nokia.com>
 
