Index: pkg.c
===================================================================
RCS file: /home/freedesktop/pkgconfig/pkg.c,v
retrieving revision 1.23
retrieving revision 1.24
diff -u -r1.23 -r1.24
--- pkg.c	26 Sep 2002 07:09:38 -0000	1.23
+++ pkg.c	2 Oct 2002 09:10:10 -0000	1.24
@@ -656,7 +656,7 @@
     }
   g_strfreev (values);
 
-  return NULL;
+  return list;
 }
 
 static void
@@ -822,6 +822,8 @@
 		      debug_spew ("Removing %s from cflags for %s\n", iter->data, pkg->key);
 		      ++count;
 		      iter->data = NULL;
+		      
+		      break;
 		    }
 		}
 	      system_dir_iter = system_dir_iter->next;
